packages/teams-js/src/public/dialog.ts

```diff
@@ -103,12 +110,30 @@ export namespace dialog {
    */
   export function submit(result?: string | object, appIds?: string | string[]): void {
     ensureInitialized(FrameContexts.content, FrameContexts.sidePanel, FrameContexts.task, FrameContexts.meetingStage);
+
     if (!isSupported()) {
       throw errorNotSupportedOnPlatform;
     }
 
-    // Send tasks.completeTask instead of tasks.submitTask message for backward compatibility with Mobile clients
-    sendMessageToParent('tasks.completeTask', [result, appIds ? (Array.isArray(appIds) ? appIds : [appIds]) : []]);
+    app
+      .getContext()
+      .then((appContext) => {
+        // if actionInfo exists in appContext, the dialog was loaded from a 3rd party app,
+        // and if submit was called with a result, throw an error because the result cannot be
+        // passed along to the desired recipient
+        if (result && appContext.actionInfo) {
+          throw errorInvalidArguments;
+        } else {
+          // Send tasks.completeTask instead of tasks.submitTask message for backward compatibility with Mobile clients
+          sendMessageToParent('tasks.completeTask', [
+            result,
+            appIds ? (Array.isArray(appIds) ? appIds : [appIds]) : [],
+          ]);
+        }
+      })
+      .catch((error) => {
+        throw error;
```

Comment:
Same as above; if this API needs to return errors asynchronously it should return a promise chain

Reply:
this was what was originally in the code, but I was asked to switch.  I assumed that this was fine since it could be wrapped in a try catch, but doing it the way you suggest is actually more consistent with the rest of the sdk